### PR TITLE
[#140] Nation Filter (connects #140)

### PR DESCRIPTION
**Pull request by razu9861**

### CODE REVIEW

**Comment by mappuji** [comment]
Do review with this step please 
https://www.youtube.com/watch?v=SmZgAULwoh4

File: src/app/nation/nation.component.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+@import '../variables.scss';
+
+.form-control{
```
Comment (lmmrssa) [inline src/app/nation/nation.component.scss]: form control is the bootstrap class. so why is there style for this specific class over here?

Comment (razu9861) [reply src/app/nation/nation.component.scss]: to give style for input field

File: src/app/nation/nation.component.html [hunk 2]
```diff
@@ -14,11 +15,13 @@ <h1 i18n>Nation List</h1>
         </tr>
       </thead>
       <tbody>
-        <tr *ngFor="let nation of nations">
+        <ng-container *ngFor="let nation of nations; let i = index">
+        <tr *ngIf="nation.doc.name.toLowerCase().includes(searchText)">
```
Comment (nirojdyola) [inline src/app/nation/nation.component.html]: Please fix indentation.

Comment (Pr0chin) [inline src/app/nation/nation.component.html]: @razu9861 "nation.doc.name.toLowerCase()" why nation name is in lowercase we can filter as if it is capital . Only letter must be in correct form.

Comment (lmmrssa) [reply src/app/nation/nation.component.html]: add tolower for searchtext as well

File: src/app/nation/nation.component.html [hunk 3]
```diff
@@ -16,11 +16,11 @@ <h1 i18n>Nation List</h1>
       </thead>
       <tbody>
         <ng-container *ngFor="let nation of nations; let i = index">
-        <tr *ngIf="nation.doc.name.toLowerCase().includes(searchText)">
-          <td>{{nation.doc.name}}</td>
-          <td>{{nation.doc.admin_name}}</td>
-          <td>{{nation.doc.nationurl}}</td>
-        </tr>
+          <tr *ngIf="nation.doc.name.toLowerCase().includes(search_text) || nation.doc.name.toUpperCase().includes(search_text) || nation.doc.name === search_text">
```
Comment (paulbert) [inline src/app/nation/nation.component.html]: I think you only need the `toLowerCase()` part, but you should also do `searchText.toLowerCase()`

File: src/app/nation/nation.component.scss [hunk 4]
```diff
@@ -1,6 +1,6 @@
 @import '../variables.scss';
 
-.form-control{
+#NationSearch{
 	float:right;
 	margin-top:-15px;
```
Comment (paulbert) [inline src/app/nation/nation.component.scss]: This makes the search bar unaligned and should be removed

File: src/app/nation/nation.component.ts [hunk 5]
```diff
@@ -24,7 +24,7 @@ export class NationComponent implements OnInit {
   message = '';
   nations = [];
   nationForm: FormGroup;
-  searchText='';
+  search_text='';
```
Comment (paulbert) [inline src/app/nation/nation.component.ts]: Camel case is more standard for JavaScript, so please revert this variable to `searchText`

File: src/app/nation/nation.component.html [hunk 6]
```diff
@@ -1,5 +1,6 @@
 <div>
   <h1 i18n>Nation List</h1>
+  <input id="NationSearch" class="form-control col-md-3 " [(ngModel)]="searchText"  i18n-placeholder placeholder="Search By Keyword">
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: pull-right class over here should do what styling for "#NationSearch" is doing

File: src/app/nation/nation.component.ts [hunk 7]
```diff
@@ -15,14 +15,16 @@ import { CustomValidators } from '../validators/custom-validators';
 import { NationValidatorService } from '../validators/nation-validator.service';
 
 @Component({
-  templateUrl: './nation.component.html'
+  templateUrl: './nation.component.html',
+  styleUrls: [ './nation.component.scss' ]
```
Comment (lmmrssa) [inline src/app/nation/nation.component.ts]: this and file will not be required once pull-right class added

File: src/app/nation/nation.component.html [hunk 8]
```diff
@@ -14,11 +15,13 @@ <h1 i18n>Nation List</h1>
         </tr>
       </thead>
       <tbody>
-        <tr *ngFor="let nation of nations">
-          <td>{{nation.doc.name}}</td>
-          <td>{{nation.doc.admin_name}}</td>
-          <td>{{nation.doc.nationurl}}</td>
-        </tr>
+        <ng-container *ngFor="let nation of nations; let i = index">
+          <tr *ngIf="nation.doc.name.toLowerCase().includes(searchText)">
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: make it nation.doc.nation.toLowerCase().includes(searchText.toLowerCase())

File: src/app/nation/nation.component.ts [hunk 9]
```diff
@@ -15,14 +15,15 @@ import { CustomValidators } from '../validators/custom-validators';
 import { NationValidatorService } from '../validators/nation-validator.service';
 
 @Component({
-  templateUrl: './nation.component.html'
+  templateUrl: './nation.component.html',
```
Comment (lmmrssa) [inline src/app/nation/nation.component.ts]: `,` at the end of last element not required

**Pull request merged**